wallet/src/de/schildbach/wallet/ui/WalletActivity.java

```diff
@@ -795,9 +796,38 @@ private void resetBlockchain() {
         alertDialog.show();
     }
 
+    // Normally OnboardingActivity will catch the non-encrypted wallets
+    // However, if OnboardingActivity does not catch it, such as after a new wallet is created,
+    // then we will catch it here.  This scenario was found during QA tests, but in a build that does
+    // not encrypt the wallet.
+
     private void checkWalletEncryptionDialog() {
         if (!wallet.isEncrypted()) {
-            EncryptKeysDialogFragment.show(false, getSupportFragmentManager());
+            log.info("the wallet is not encrypted");
+            analytics.logError(new Exception("the wallet is not encrypted / OnboardingActivity"),
+                    "no other details are available without the user submitting a report");
+            AdaptiveDialog dialog = AdaptiveDialog.custom(R.layout.dialog_adaptive,
+                    R.drawable.ic_error,
+                    getString(R.string.wallet_encryption_error_title),
+                    getString(R.string.wallet_not_encrypted_error_message),
+                    getString(R.string.button_cancel),
+                    getString(R.string.button_ok)
+            );
+            dialog.show(this, reportIssue -> {
+                if (reportIssue != null) {
```

Comment:
Is dismissing this dialog without choosing a button is the same as choosing "Cancel"? In this case, the null result should be accounted for. A three-state result (true, false, null) is made to accommodate specific negative action in case it is different from dismissing.

Reply:
Dismissing is the same as pressing cancel.  Though I made a change to this dialog to make it not cancel-able (a button must be pressed).  If that is the case, then there should be no `null` result, right?

Reply:
Yes, this should do.